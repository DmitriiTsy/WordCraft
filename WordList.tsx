import React from 'react';
import { FlatList, Text } from 'react-native';

interface WordListProps {
  validWords: string[];
}

const WordList: React.FC<WordListProps> = ({ validWords }) => {
  return (
    // Render the list of valid words
    <FlatList
      data={validWords}
      renderItem={({ item, index }) => <Text key={index}>{item}</Text>}
      style={{ marginTop: 20 }}
    />
  );
};

export default WordList;
