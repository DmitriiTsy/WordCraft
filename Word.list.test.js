import React from 'react';
import { FlatList, Text } from 'react-native';
import renderer from 'react-test-renderer';
import WordList from './WordList'; // Adjust the import path based on your project structure

describe('WordList Component', () => {
  const validWords = ['apple', 'banana', 'cherry'];

  it('renders correctly with validWords', () => {
    const tree = renderer.create(<WordList validWords={validWords} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an empty list with no validWords', () => {
    const tree = renderer.create(<WordList validWords={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct number of words', () => {
    const component = renderer.create(<WordList validWords={validWords} />);
    const textComponents = component.root.findAllByType(Text);
    expect(textComponents.length).toBe(validWords.length);
  });
});
