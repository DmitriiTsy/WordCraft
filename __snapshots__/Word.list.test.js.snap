// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordList Component renders an empty list with no validWords 1`] = `
<RCTScrollView
  data={[]}
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  style={
    {
      "marginTop": 20,
    }
  }
  viewabilityConfigCallbackPairs={[]}
>
  <View />
</RCTScrollView>
`;

exports[`WordList Component renders correctly with validWords 1`] = `
<RCTScrollView
  data={
    [
      "apple",
      "banana",
      "cherry",
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  style={
    {
      "marginTop": 20,
    }
  }
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        apple
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        banana
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        cherry
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
